client/sniffer/src/main/java/org/elasticsearch/client/sniff/ElasticsearchNodesSniffer.java

```diff
@@ -140,16 +144,19 @@ private static Node readNode(String nodeId, JsonParser parser, Scheme scheme) th
         Set<HttpHost> boundHosts = new HashSet<>();
         String name = null;
         String version = null;
-        String fieldName = null;
-        // Used to read roles from 5.0+
+        /*
+         * Multi-valued attributes come with key = `real_key.index` and we
+         * unflip them after reading them because we can't rely on the that
+         * they arive in the result object.
```

Comment:
arive?

Reply:
Sorry! We can't rely on the order that they arrive. Proxies could jumble them up and we *should* still work. 2.x looks like it jumbles them up for fun.